Convert Project to Kotlin

**Is your feature request related to a problem? Please describe.**
Kotlin is our primary language but 40% of the codebase is still in java, this ticket aims to get that number to 0%. If you want to contribute to the project but are finding it diffcult to find a suitable ticket of a manageable size then this is for you!

**Describe the solution you'd like**
Comment here that you want to convert a class (I recommend starting with small ones) with the class name to claim it. Then make a ticket and start working on it. Use `ctrl+alt+shift+k` to automatically convert the file and then improve it to be more idiomatic kotlin.

**Additional context**
If you have never worked in kotlin before the [koans](https://kotlinlang.org/docs/tutorials/koans.html) (and the kotlin docs in general) are a good way to get used to the features. There is also support from kiwix members if you want to open a PR in draft

@kelson42 @gouri-panda  WHENEVER I AM DOING A COMMIT ON YOUR  CLONED PROJECT I AM GETTING "commit failed with error" and the commit is failing . CAN YOU PLEASE HELP ME OUT ?

DESCRIPTION OF ERROR :

Running lint... Starting a Gradle Daemon, 1 incompatible Daemon could not be reused, use --status for details > Task :buildSrc:compileKotlin FAILED FAILURE: Build failed with an exception. * What went wrong: Execution failed for task ':buildSrc:compileKotlin'. > Kotlin could not find the required JDK tools in the Java installation 'C:\Program Files\Java\jre1.8.0_301' used by Gradle. Make sure Gradle is running on a JDK, not JRE. * Try: Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights. * Get more help at https://help.gradle.org BUILD FAILED in 10s > Task :buildSrc:compileKotlin FAILED FAILURE: Build failed with an exception. * What went wrong: Execution failed for task ':buildSrc:compileKotlin'. > Kotlin could not find the required JDK tools in the Java installation 'C:\Program Files\Java\jre1.8.0_301' used by Gradle. Make sure Gradle is running on a JDK, not JRE. * Try: Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights. * Get more help at https://help.gradle.org BUILD FAILED in 2s Static analysis found violations and attempted to autofix, please commit these autoformat changes

@gouri-panda  yes I did the following changes which you asked but things are not committing yet. but the error has changed from before 

description of error:-
Starting a Gradle Daemon, 1 incompatible and 4 stopped Daemons could not be reused, use --status for details FAILURE: Build failed with an exception. * What went wrong: Unable to start the daemon process. This problem might be caused by incorrect configuration of the daemon. For example, an unrecognized jvm option is used. Please refer to the User Manual chapter on the daemon at https://docs.gradle.org/6.2/userguide/gradle_daemon.html Process command line: C:\Program Files\Java\jdk1.8.0_301\bin\java.exe -Xmx4096m -Dfile.encoding=windows-1252 -Duser.country=IN -Duser.language=en -Duser.variant -cp C:\Users\risha\.gradle\wrapper\dists\gradle-6.2-all\dvufqs6kielxeao781pmk5huj\gradle-6.2\lib\gradle-launcher-6.2.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 6.2 Please read the following process output to find out more: ----------------------- * Try: Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights. * Get more help at https://help.gradle.org FAILURE: Build failed with an exception. * What went wrong: java.net.BindException: Address already in use: Cannot bind * Try: Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights. * Get more help at https://help.gradle.org Static analysis found violations and attempted to autofix, please commit these autoformat changes

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/di/qualifiers/Computation.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/di/qualifiers/Computation.kt
@@ -1,6 +1,6 @@
 /*
  * Kiwix Android
- * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * Copyright (c) 2022 Kiwix <android.kiwix.org>
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
  * the Free Software Foundation, either version 3 of the License, or
@@ -16,13 +16,10 @@
  *
  */
 
-package org.kiwix.kiwixmobile.core.di.qualifiers;
+package org.kiwix.kiwixmobile.core.di.qualifiers
 
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
-import javax.inject.Qualifier;
+import javax.inject.Qualifier
 
 @Qualifier
-@Retention(RetentionPolicy.RUNTIME)
-public @interface Computation {
-}
+@kotlin.annotation.Retention
+annotation class Computation

